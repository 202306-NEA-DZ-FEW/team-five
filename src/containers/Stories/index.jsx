import AOS from "aos";
import { useTranslation } from "next-i18next";
import React, { useEffect } from "react";

import "aos/dist/aos.css";

function Stories() {
    const { t } = useTranslation("stories");

    useEffect(() => {
        AOS.init();
    }, []);

    return (
        <div className='py-12 sm:py-12 max-w-screen-xl mx-auto'>
            <div className='mx-auto max-w-7xl px-6 lg:px-8'>
                <div className='mb-5'>
                    <h2 className='text-center text-3xl font-semibold tracking-tight text-sky-600 sm:text-4xl'>
                        {t("life")}
                    </h2>
                    <p className='text-center mt-4 text-lg leading-8 text-gray-600 mb-5'>
                        {t("we")}
                    </p>
                </div>
                <ul
                    role='list'
                    className='mx-auto mt-5 grid max-w-2xl grid-cols-1 gap-x-8 gap-y-16 sm:grid-cols-2 lg:mx-0 lg:max-w-none lg:grid-cols-3'
                    data-aos='fade-up'
                    data-aos-duration='1000'
                >
                    <li
                        key='name1'
                        className='group cursor-pointer transform transition-transform hover:scale-105 hover:bg-gray-200'
                    >
                        <img
                            className='aspect-[3/2] w-full object-cover'
                            src='storie01.png'
                            alt=''
                        />
                        <h3 className='mt-6 text-xs font-semibold leading-8 tracking-tight text-gray-600'>
                            {t("story")}
                        </h3>
                        <p className='text-xl leading-7 text-sky-600'>
                            {t("blog1")}
                        </p>
                    </li>
                    <li
                        key='name2'
                        className='group cursor-pointer transform transition-transform hover:scale-105 hover:bg-gray-200'
                    >
                        <img
                            className='aspect-[3/2] w-full object-cover'
                            src='storie02.png'
                            alt=''
                        />
                        <h3 className='mt-6 text-xs font-semibold leading-8 tracking-tight text-gray-600'>
                            {t("story")}
                        </h3>
                        <p className='text-xl leading-7 text-sky-600'>
                            {t("blog2")}
                        </p>
                    </li>
                    <li
                        key='name3'
                        className='group cursor-pointer transform transition-transform hover:scale-105 hover:bg-gray-200'
                    >
                        <img
                            className='aspect-[3/2] w-full object-cover'
                            src='storie03.png'
                            alt=''
                        />
                        <h3 className='mt-6 text-xs font-semibold leading-8 tracking-tight text-gray-600'>
                            {t("story")}
                        </h3>
                        <p className='text-xl leading-7 text-sky-600'>
                            {t("blog3")}
                        </p>
                    </li>
                </ul>
            </div>
        </div>
    );
}

export default Stories;
