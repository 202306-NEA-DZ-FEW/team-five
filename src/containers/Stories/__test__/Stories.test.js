import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import Stories from "..";

jest.mock("next/router", () => ({
    useRouter: () => ({
        route: "/",
        pathname: "",
        query: {},
        asPath: "",
    }),
}));

jest.mock("react-i18next", () => ({
    useTranslation: () => ({ t: (key) => key, i18n: { language: "en" } }),
}));

const mockStore = configureMockStore([]);
const store = mockStore({
    userInfo: null,
});

describe("Blog Component", () => {
    it("should render correctly", () => {
        const component = renderer.create(
            <Provider store={store}>
                <Stories />
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
