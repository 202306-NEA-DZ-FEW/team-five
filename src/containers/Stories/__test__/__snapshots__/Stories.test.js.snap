// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Component should render correctly 1`] = `
<div
  className="py-12 sm:py-12 max-w-screen-xl mx-auto"
>
  <div
    className="mx-auto max-w-7xl px-6 lg:px-8"
  >
    <div
      className="mb-5"
    >
      <h2
        className="text-center text-3xl font-semibold tracking-tight text-sky-600 sm:text-4xl"
      >
        life
      </h2>
      <p
        className="text-center mt-4 text-lg leading-8 text-gray-600 mb-5"
      >
        we
      </p>
    </div>
    <ul
      className="mx-auto mt-5 grid max-w-2xl grid-cols-1 gap-x-8 gap-y-16 sm:grid-cols-2 lg:mx-0 lg:max-w-none lg:grid-cols-3"
      data-aos="fade-up"
      data-aos-duration="1000"
      role="list"
    >
      <li
        className="group cursor-pointer transform transition-transform hover:scale-105 hover:bg-gray-200"
      >
        <img
          alt=""
          className="aspect-[3/2] w-full object-cover"
          src="storie01.png"
        />
        <h3
          className="mt-6 text-xs font-semibold leading-8 tracking-tight text-gray-600"
        >
          story
        </h3>
        <p
          className="text-xl leading-7 text-sky-600"
        >
          blog1
        </p>
      </li>
      <li
        className="group cursor-pointer transform transition-transform hover:scale-105 hover:bg-gray-200"
      >
        <img
          alt=""
          className="aspect-[3/2] w-full object-cover"
          src="storie02.png"
        />
        <h3
          className="mt-6 text-xs font-semibold leading-8 tracking-tight text-gray-600"
        >
          story
        </h3>
        <p
          className="text-xl leading-7 text-sky-600"
        >
          blog2
        </p>
      </li>
      <li
        className="group cursor-pointer transform transition-transform hover:scale-105 hover:bg-gray-200"
      >
        <img
          alt=""
          className="aspect-[3/2] w-full object-cover"
          src="storie03.png"
        />
        <h3
          className="mt-6 text-xs font-semibold leading-8 tracking-tight text-gray-600"
        >
          story
        </h3>
        <p
          className="text-xl leading-7 text-sky-600"
        >
          blog3
        </p>
      </li>
    </ul>
  </div>
</div>
`;
